elastic/kibana
# Reapply "[Response Ops][Alerting] Backfill actions schema changes for intermediate release

## Summary

Redoing [this PR](https://github.com/elastic/kibana/pull/203184) which had to be [reverted](https://github.com/elastic/kibana/pull/204218). This should not be merged until [this update to the task manager v1 schema](https://github.com/elastic/kibana/pull/204413) is released.

## To verify
1. Set `xpack.task_manager.unsafe.exclude_task_types: ['ad_hoc_run-backfill', 'actions:*']` in your Kibana config. 
2. Run Kibana on main and create some detection rules that run frequently, with actions.
3. Schedule a manual run for your detection rules.
    - Because of the config, the `action_task_params` SO and the `ad_hoc_run_task_params` SO will get written but not read yet.
4. Remove the `exclude_task_types` config and "upgrade" to this PR branch and verify that rules continue to run and that the actions are triggered and the manual rule runs go through
5. Re-add the `exclude_task_types` config and let the rule run again to schedule action. Schedule another manual rule run.
6. Remove the `exclude_task_types` config and "downgrade" back to main and verify that rules continue to run, the action gets triggered and manual rule runs go through.
